packages/calling-stateful-client/src/CallDeclarativeCommon.ts

```diff
@@ -82,7 +82,10 @@ export abstract class ProxyCallCommon implements ProxyHandler<CallCommon> {
                 const ret = await captionsFeature.startCaptions(...args);
 
                 this._context.setIsCaptionActive(target.id, true);
-                this._context.setSelectedSpokenLanguage(target.id, args[0]?.spokenLanguage ?? 'en-us');
+                if (args[0]?.spokenLanguage && args[0]?.spokenLanguage !== '') {
```

Comment:
We shouldn't do it here, we should modify the behavior when you click the button, not the behavior of the function itself, functions should stay what they mean to be

+1 to keep `this._context.setSelectedSpokenLanguage(target.id, args[0]?.spokenLanguage);` here

For calling `await captionsFeature.setSpokenLanguage(args[0].spokenLanguage);` it should be happening in where onStartCaption function in the component